core/modules/node/node.types.inc

```diff
@@ -130,6 +130,92 @@ function node_type_form($form, &$form_state, $type = NULL) {
+  $auto_permission_roles = array();
+  if ($default_admin_role && $admin_role = user_role_load($default_admin_role)) {
+    $auto_permission_roles['%admin'] = $admin_role->label;
+  }
+  if ($default_editor_role && $editor_role = user_role_load($default_editor_role)) {
+    $auto_permission_roles['%editor'] = $editor_role->label;
+  }
+  // Handle the case where the same role has been configured to be both the
+  // default admin as well as the default editor.
+  $auto_permission_roles = array_unique($auto_permission_roles);
+  // Prepare the $auto_permission_roles array for format_plural() when there's
+  // only one role.
+  if (count($auto_permission_roles) == 1) {
+    $one_role = reset($auto_permission_roles);
+    unset($auto_permission_roles);
+    $auto_permission_roles['%role'] = $one_role;
+  }
+
+  $permissions_assigned = FALSE;
+  $content_type_permissions = node_type_form_permissions($type->type);
+  foreach ($content_type_permissions['checkboxes'] as $role => $checkboxes) {
+    if (!empty($checkboxes['#default_value'])) {
+      // At least one permission related to this content type is assigned to at
+      // least one role.
+      $permissions_assigned = TRUE;
+      break;
+    }
+  }
+
+  // Edge case, where both the default admin role and the default content editor
+  // role have been disabled in in admin/config/people/roles, while at the same
+  // time no other role has been granted any permission associated with this
+  // content type. We show this warning for both new and existing content types.
+  if (!$permissions_assigned) {
+    $permissions_message = t('No permissions assigned for this content type. Content of this type may not be able to be created, updated or deleted until permissions have been configured appropriately.');
+    $permissions_message_type = 'warning';
+  }
+  // Show some info messages if automatic permission assignment is enabled for
+  // at least one of the default admin/editor roles. We only show these when
+  // creating a new content type.
+  if (!empty($auto_permission_roles) && empty($type->type)) {
+    $permissions_message = format_plural(
+      count($auto_permission_roles),
+      'The %role role has been assigned permissions to create, edit and delete any content of this type.',
+      'The %admin and %editor roles have been assigned permissions to create, edit and delete any content of this type.',
+      $auto_permission_roles
+    );
+    $permissions_message_type = 'info';
+  }
+
+  if (isset($permissions_message)) {
+    if (user_access('administer permissions')) {
+      $permissions_message .= ' ' . t('You can make any necessary adjustments in the <em>Permissions</em> tab below.');
+    }
+    else {
+      $user_module_permissions = module_invoke('user', 'permission');
+      $permissions_message .= ' ' . t('Only roles with the %permission permission can make any necessary adjustments.', array('%permission' => $user_module_permissions['administer permissions']['title']));
```

Comment:
Also a follow-up issue, there should be a way to get a permission label.

Reply:
Done: https://github.com/backdrop/backdrop-issues/issues/6097